src/bin/dual-iir.rs

```diff
@@ -106,159 +140,47 @@ const APP: () = {
-                                    c.resources.iir_ch.lock(|iir_ch| {
-                                        if req.channel > 1 {
-                                            return Err(());
-                                        }
-
-                                        iir_ch[req.channel as usize][IIR_CASCADE_LENGTH-1] = req.iir;
-
-                                        Ok::<server::IirRequest, ()>(req)
-                                    })
-                                }),
-                                "stabilizer/iir_b1/state": server::IirRequest,(|req: server::IirRequest| {
-                                    c.resources.iir_ch.lock(|iir_ch| {
-                                        if req.channel > 1 {
-                                            return Err(());
-                                        }
-
-                                        iir_ch[req.channel as usize][IIR_CASCADE_LENGTH-1] = req.iir;
-
-                                        Ok::<server::IirRequest, ()>(req)
-                                    })
-                                }),
-                                "stabilizer/afe0/gain": hardware::AfeGain, (|gain| {
-                                    c.resources.afes.0.set_gain(gain);
-                                    Ok::<(), ()>(())
-                                }),
-                                "stabilizer/afe1/gain": hardware::AfeGain, (|gain| {
-                                    c.resources.afes.1.set_gain(gain);
-                                    Ok::<(), ()>(())
-                                })
-                            ]
-                        )
-                    });
+                miniconf::Action::Continue => {
+                    if sleep {
+                        cortex_m::asm::wfi();
+                    }
                 }
-            }
-
-            let sleep = match c.resources.net_interface.poll(
-                &mut sockets,
-                smoltcp::time::Instant::from_millis(time as i64),
-            ) {
-                Ok(changed) => !changed,
-                Err(smoltcp::Error::Unrecognized) => true,
-                Err(e) => {
-                    info!("iface poll error: {:?}", e);
-                    true
+                miniconf::Action::CommitSettings => {
+                    c.spawn.settings_update().unwrap()
                 }
-            };
-
-            if sleep {
-                cortex_m::asm::wfi();
             }
         }
     }
 
+    #[task(priority = 1, resources=[mqtt_interface, afes, iir_ch, iir_state])]
+    fn settings_update(mut c: settings_update::Context) {
+        let settings = &c.resources.mqtt_interface.settings;
+
+        // Update the IIR channels.
+        c.resources.iir_ch.lock(|iir| *iir = settings.iir_ch);
+
+        // Update the IIR states.
+        c.resources.iir_state.lock(|iir| *iir = settings.iir_state);
```

Comment:
@jordens You had mentioned that we don't want to update the IIR channels all the time - would we always want to update the `iir_state` and `iir_channel` simultaneously? Also, would we want to update CH0/1 independently?

Reply:
Yes. Only update the `iir_state` if that is explicitly flagged.

Also, this is somewhat related to https://github.com/quartiq/stabilizer/issues/70
Let me know if that description isn't clear.

Reply:
The more I think about it, for now it's ok to never update `iir_state`. Let's keep it out of settings.